contracts/ConvictionVoting.sol

```diff
@@ -221,11 +246,36 @@ contract ConvictionVoting is AragonApp, TokenManagerHook {
         require(proposal.convictionLast > calculateThreshold(proposal.requestedAmount), ERROR_INSUFFICIENT_CONVICION);
 
         proposal.proposalStatus = ProposalStatus.Executed;
-        vault.transfer(requestToken, proposal.beneficiary, proposal.requestedAmount);
+
+        if (proposal.evmScript.length == 0) {
+            vault.transfer(address(requestToken), proposal.beneficiary, proposal.requestedAmount);
+        } else {
+            _executeScriptWithTransfer(proposal);
+        }
 
         emit ProposalExecuted(_proposalId, proposal.convictionLast);
     }
 
+    function _executeScriptWithTransfer(Proposal storage proposal) internal {
+        uint256 balanceBeforeExecution;
+
+        if (proposal.requiresApproval) {
```

Comment:
Wonder if the `proposal.requestedAmount` is 0 should we skip this if..else block of code?